doc/java-generation-from-CRD.md

```diff
@@ -0,0 +1,123 @@
+# Java generation from CRD (preview)
+
+This document shows how to generate java sources from CRD manifests.
+
+## Quick start CLI
+
+The Java generator CLI is available for download on Sonatype at the link:
+
+```
+https://oss.sonatype.org/content/repositories/releases/io/fabric8/java-generator-cli/<version>/java-generator-cli-<version>.sh
+```
+
+you can get it working locally with few lines:
+
+```bash
+export VERSION=<version>
+wget https://oss.sonatype.org/content/repositories/releases/io/fabric8/java-generator-cli/$VERSION/java-generator-cli-$VERSION.sh
+chmod a+x java-generator-cli-$VERSION.sh
+./java-generator-cli-$VERSION.sh --version
+```
+
+Alternatively, if you already have [jbang](https://www.jbang.dev/) installed, you can run the CLI using it:
+
+```bash
+jbang io.fabric8:java-generator-cli:<version>
+```
+
+## Quick start Maven
+
+The Java generator Maven plugin can be used from your project `pom.xml` adding a section like:
+
+```xml
+<plugin>
+  <groupId>io.fabric8</groupId>
+  <artifactId>java-generator-maven-plugin</artifactId>
+  <version><version></version>
```

Comment:
```suggestion
  <version>${kubernetes-client.version}</version>
```
Maybe this line is confusing.

I'd replace this `<version>` value since the resulting XML is invalid and it's unclear.